### PR TITLE
feat(JOML): get rid of `Region3i`

### PR DESCRIPTION
**Pull request by jdrueckert**

### CODE REVIEW

File: src/main/java/org/terasology/cities/deco/ColumnDecoration.java [hunk 1]
```diff
@@ -39,17 +40,17 @@
      * @param sides the facing sides of the blocks
      * @param basePos the window position
      */
-    public ColumnDecoration(List<BlockType> blocks, List<Side> sides, BaseVector3i basePos) {
+    public ColumnDecoration(List<BlockType> blocks, List<Side> sides, Vector3ic basePos) {
         Preconditions.checkArgument(blocks.size() == sides.size(), "blockCount != sideCount");
         this.blocks = blocks;
         this.sides = sides;
-        this.pos = ImmutableVector3i.createOrUse(basePos);
+        this.pos = basePos;
```
Comment (pollend) [inline src/main/java/org/terasology/cities/deco/ColumnDecoration.java]:
```suggestion
        this.pos.set(basePos);
```

File: src/main/java/org/terasology/cities/deco/ColumnDecoration.java [hunk 2]
```diff
@@ -30,7 +31,7 @@
  */
 public class ColumnDecoration implements Decoration {
 
-    private final ImmutableVector3i pos;
+    private final Vector3ic pos;
```
Comment (pollend) [inline src/main/java/org/terasology/cities/deco/ColumnDecoration.java]:
```suggestion
    private final Vector3ic pos = new Vector3i();
```

File: src/main/java/org/terasology/cities/deco/ColumnDecoration.java (hunk 1 above)
Comment (pollend) [reply src/main/java/org/terasology/cities/deco/ColumnDecoration.java]: write to a pre-declared one. makes it more immutable?

File: src/main/java/org/terasology/cities/deco/SingleBlockDecoration.java [hunk 3]
```diff
@@ -35,16 +36,16 @@
      * @param pos the window position
      * @param side the direction of the decoration
      */
-    public SingleBlockDecoration(BlockType type, BaseVector3i pos, Side side) {
+    public SingleBlockDecoration(BlockType type, Vector3ic pos, Side side) {
         this.type = type;
         this.side = side;
-        this.pos = ImmutableVector3i.createOrUse(pos);
+        this.pos = pos;
```
Comment (pollend) [inline src/main/java/org/terasology/cities/deco/SingleBlockDecoration.java]:
```suggestion
        this.pos.set(pos);
```

File: src/main/java/org/terasology/cities/deco/SingleBlockDecoration.java [hunk 4]
```diff
@@ -26,7 +27,7 @@
  */
 public class SingleBlockDecoration implements Decoration {
 
-    private final ImmutableVector3i pos;
+    private final Vector3ic pos;
```
Comment (pollend) [inline src/main/java/org/terasology/cities/deco/SingleBlockDecoration.java]:
```suggestion
    private final Vector3ic pos = new Vector3i();
```

**Review by pollend — CHANGES_REQUESTED** [review]

**Review by pollend — DISMISSED** [review]

**Review by pollend — APPROVED** [review]

**Review by skaldarnar — APPROVED** [review]

**Pull request merged**